src/Core/IdentityServer/CustomTokenRequestValidator.cs

```diff
@@ -95,7 +95,8 @@ protected async override Task<(User, bool)> ValidateContextAsync(CustomTokenRequ
             if (context.Result.ValidatedRequest.GrantType == "client_credentials")
             {
                 if (user.UsesKeyConnector) {
-                    // KeyConnectorUrl is configured in the CLI client, just disable master password reset
+                    // KeyConnectorUrl is configured in the CLI client, we just need to tell the client to use it
+                    context.Result.CustomResponse["ApiUseKeyConnector"] = true;
```

Comment:
Should we remove `Api` from this response? Feels like we should use this value for the other clients as well.

Reply:
The reason I put `Api` in there is that it's only used when logging in with an `apikey`. The other clients simply don't need this flag - they can just use the `KeyConnectorUrl` if it's not null, no need to add to the payload in those cases. I didn't want people thinking this is a general purpose flag that they have to check in other circumstances.

The alternative would be to re-use the existing `KeyConnectorUrl` property with some special string that CLI can test for, but that felt like misusing a string to send a boolean value.